src/Enzo/hydro-mhd/EnzoMethodMHDVlct.cpp

```diff
@@ -35,12 +35,86 @@ static str_vec_t concat_str_vec_(const str_vec_t& vec1, const str_vec_t& vec2){
 
 //----------------------------------------------------------------------
 
-EnzoMethodMHDVlct::EnzoMethodMHDVlct (std::string rsolver,
-				      std::string time_scheme,
-                                      std::string recon_name,
-				      double theta_limiter,
-				      std::string mhd_choice,
-				      bool store_fluxes_for_corrections)
+static std::pair<std::string, EnzoMHDIntegratorStageArgPack*>
+parse_mhdchoice_pack_pair_(ParameterGroup p)
+{
+  // is_defined is a function that indicates if a user defined a parameter
+  auto is_defined = [&](const std::string& param_name) -> bool
+  { return p.param(param_name) != nullptr; };
+
+  // parse time-scheme & reconstruct-method (they affect backwards compat.)
+  const std::string time_scheme = p.value_string("time_scheme", "vl");
+  std::vector<std::string> recon_names;
+  if (time_scheme == "euler") {
+    recon_names = {p.value_string("reconstruct_method", "plm")};
+    WARNING("parse_mhdchoice_pack_pair_",
+            "\"euler\" temporal integration exists for debugging purposes. It "
+            "has not been rigorously tested. USE WITH CAUTION");
+  } else if (time_scheme == "vl") {
+    // ALWAYS use nearest-neighbor reconstruction for partial timestep!
+    recon_names = {"nn", p.value_string("reconstruct_method", "plm")};
+  } else {
+    ERROR1("parse_mhdchoice_pack_pair_",
+           "\"%s:time_scheme\" must be \"vl\" or \"euler\"",
+           p.get_group_path());
+  }
+
+  // backwards compatibilty: deprecated half/full_dt_reconstruct_method params
+  if (is_defined("half_dt_reconstruct_method") ||
+      is_defined("full_dt_reconstruct_method")) {
+
+    if (is_defined("time_scheme") || is_defined("reconstruct_method")) {
+      ERROR1("parse_mhdchoice_pack_pair_",
+             "In the \"%s\" parameter-group, the deprecated parameters, "
+             "\"half_dt_reconstruct_method\" & \"full_dt_reconstruct_method\", "
+             "can't be specified when \"time_scheme\" or "
+             "\"reconstruct_method\" is specified.",
+             p.get_group_path().c_str());
+    }
+
+    if (p.value_string("half_dt_reconstruct_method", "nn") != "nn") {
+      // it never made ANY sense to allow the half timestep of the VL+CT
+      // algorithm to use anything other than the "nn" choice. (The only reason
+      // it was ever an option was due to a misunderstanding early on)
+      ERROR1("parse_mhdchoice_pack_pair_",
+             "The deprecated parameter, \"%s:half_dt_reconstruct_method\", "
+             "can't have any value other than \"nn\"",
+             p.get_group_path().c_str());
+    }
+
+    if (is_defined("full_dt_reconstruct_method")) {
+      recon_names[1] = p.value_string("full_dt_reconstruct_method", "plm");
```

Comment:
Is assuming `recon_names` has two entries going to be an issue? e.g. is it possible for the `time_scheme` to be Euler and reach this piece of code? Doesn't seem so, as `time_scheme` must be undefined to reach this block and it's default value is `vl`, but wanted to double check :)

Reply:
Thanks for pointing this out! 

I definitely could have made a mistake here, but I think this is ok:
- On [line 46](https://github.com/enzo-project/enzo-e/blob/ba857eef6d8f78958915fd074dddb3d46d9f994f/src/Enzo/hydro-mhd/EnzoMethodMHDVlct.cpp#L46), logic is in place that assigns the `time_scheme` variable a value of `"vl"` if the `Method:mhd_vlct:time_scheme` parameter was undefined.
- The logic here is indeed a little convoluted (basically, I wanted to force people to stop using the deprecated parameters if they started using the newer parameters)

I think I'll submit a new PR to remove this deprecated parameter completely. I was tempted to do that sooner (since I'm the main/only user of this solver right now), but I didn't want to keep the removal out of this PR.

Reply:
I agree the current program flow doesn't look like it should introduce any errors (at least from what I see within this PR) so it's fine if you want to refactor in another PR. It's probably best to keep this PR a more straightforward port.